session-helper/flatpak-session-helper.c

```diff
@@ -401,6 +404,48 @@ handle_host_command_signal (FlatpakDevelopment    *object,
   return TRUE;
 }
 
+static void
+name_owner_changed (GDBusConnection *connection,
+                    const gchar     *sender_name,
+                    const gchar     *object_path,
+                    const gchar     *interface_name,
+                    const gchar     *signal_name,
+                    GVariant        *parameters,
+                    gpointer         user_data)
+{
+  const char *name, *from, *to;
+
+  g_variant_get (parameters, "(&s&s&s)", &name, &from, &to);
+
+  if (name[0] == ':' &&
+      strcmp (name, from) == 0 &&
+      strcmp (to, "") == 0)
+    {
+      GHashTableIter iter;
+      PidData *pid_data = NULL;
+      GList *list = NULL, *l;
+
+      g_hash_table_iter_init (&iter, client_pid_data_hash);
+      while (g_hash_table_iter_next (&iter, NULL, (gpointer *)pid_data))
+        {
+          if (pid_data->watch_bus && g_str_equal (pid_data->client, name))
+            list = g_list_prepend (list, pid_data);
+        }
+
+      for (l = list; l; l = l->next)
+        {
+          pid_data = l->data;
+          kill (pid_data->pid, SIGKILL);
```

Comment:
Is sigkill the best thing to send here? I could imagine situations where you want SIGINT so you can do some cleanup. Maybe this could be a separate flag?

Reply:
I've make it use killpg (pid, SIGINT) now. We can revisit when users find they need sth. else.